src/interceptors/XMLHttpRequest/XMLHttpRequestOverride.ts

```diff
@@ -532,5 +533,17 @@ export const createXMLHttpRequestOverride = (
         req.addEventListener(name, listener)
       })
     }
+
+    propagateHeaders(
+      req: XMLHttpRequest,
+      headers: Record<string, string | string[]>
+    ) {
+      Object.entries(headers).forEach(([key, value]) => {
+        req.setRequestHeader(
+          key,
+          Array.isArray(value) ? value.join(',') : value
```

Comment:
Could you please use `flattenHeadersObject` from [`headers-utils`](https://github.com/mswjs/headers-utils) for this?

```js
const flatHeaders = flattenHeadersObject(headers)
Object.entries(flatHeaders).forEach(...)
```